specs/WebResourceResponseReceived.md

```diff
@@ -0,0 +1,232 @@
+    /// response objects are ignored.
+    interface ICoreWebView2WebResourceResponseReceivedEventHandler : IUnknown
+    {
+        /// Called to provide the implementer with the event args for the
+        /// corresponding event.
+        HRESULT Invoke(
+            [in] ICoreWebView2* sender,
+            [in] ICoreWebView2WebResourceResponseReceivedEventArgs* args);
+    }
+
+    /// Completion handler for PopulateResponseContent async method. It's invoked
+    /// when the Content stream of the Response of a WebResourceResponseReceieved
+    /// event is available.
+    interface ICoreWebView2WebResourceResponseReceivedEventArgsPopulateResponseContentCompletedHandler : IUnknown
+    {
+        /// Called to provide the implementer with the completion status
+        /// of the corresponding asynchronous method call.
+        HRESULT Invoke([in] HRESULT errorCode);
+    }
+
+    /// Event args for the WebResourceResponseReceived event. Will contain the
+    /// request as it was sent and the response as it was received.
+    /// Note: To get the response content stream, first call PopulateResponseContent
+    /// and wait for the async call to complete, otherwise the content stream object
+    /// returned will be null.
+    interface ICoreWebView2WebResourceResponseReceivedEventArgs : IUnknown
+    {
+        /// Web resource request object. Any modifications to this object will be ignored.
+        [propget] HRESULT Request([out, retval] ICoreWebView2WebResourceRequest** request);
+        /// Web resource response object. Any modifications to this object
+        /// will be ignored.
+        [propget] HRESULT Response([out, retval] ICoreWebView2WebResourceResponse** response);
+
+        /// Async method to ensure that the Content property of the response contains the actual response body content.
+        /// If this method is being called again before a first call has completed, all handlers are invoked at the same time.
+        /// If this method is being called after a first call has completed, the handler is invoked immediately.
+        HRESULT PopulateResponseContent(ICoreWebView2WebResourceResponseReceivedEventArgsPopulateResponseContentCompletedHandler* handler);
+    }
+}
+```
+
+## WinRT
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    runtimeclass CoreWebView2
+    {
+        // ...
+
+        /// WebResourceResponseReceived event is raised after the WebView has received and processed the response for a WebResource request.
+        /// The event args include the WebResourceRequest as committed and the WebResourceResponse received,
+        /// including any additional headers added by the network stack that were not be included as part of
+        /// the associated WebResourceRequested event, such as Authentication headers.
+        event Windows.Foundation.TypedEventHandler<CoreWebView2, CoreWebView2WebResourceResponseReceivedEventArgs> WebResourceResponseReceived;
```

Comment:
IDL says it's a TypedEventHandler (sender is strongly typed), but sample implies that it's an EventHandler (sender is just an `object`). Who's right?

Reply:
Should update the sample to follow the IDL

Reply:
`TypedEventHandler`.